### PR TITLE
REFACT: Avoid assert calls and try except pass pattern

### PR DESCRIPTION
**Pull request by SMoraisAnsys**
Updating lines of code considered as bad practice because they may have side effects that are not directly taken into account.

Refact changes:
- Avoid using `assert` as such calls are removed with compiling to optimized byte code. Instead, we now raise an exception (`ValueError` or `KeyError` or `RuntimeError`).
- Avoid try: except: pass pattern. Instead, we now log something to info (for public methods) or debug (for other methods).
- Adding `# nosec` for `subprocess` imports and calls (from what I've seen they shouldn't be a security problem).

Note: Some code refactoring was performed when it made sense.

### CODE REVIEW

File: pyaedt/generic/general_methods.py [hunk 1]
```diff
@@ -237,7 +237,7 @@ def wrapper(*args, **kwargs):
         except GrpcApiError as e:
             _exception(sys.exc_info(), user_function, args, kwargs, "AEDT grpc API call Error")
             raise_exception(e)
-        except BaseException as e:
+        except (BaseException, ValueError, KeyError, RuntimeError) as e:
```
Comment (SMoraisAnsys) [inline pyaedt/generic/general_methods.py:240]:
```suggestion
        except BaseException as e:
```

File: pyaedt/aedt_logger.py [hunk 2]
```diff
@@ -579,7 +579,8 @@ def _log_on_dekstop(self, message_type, message_text, level=None, proj_name=None
         if not level:
             level = "Design"
 
-        assert level in message_levels, "Message level must be `Design', 'Project', or 'Global'."
+        if level not in message_levels:
+            raise ValueError("Message level must be 'Design', 'Project' or 'Global'.")
```
Comment (PipKat) [inline pyaedt/aedt_logger.py:583]:
```suggestion
            raise ValueError("Message level must be 'Design', 'Project', or 'Global'.")
```
We always use an Oxford comma, which say to place a comma before an "or" or an "and" in a list of three or more items. What was originally there was correct.

File: pyaedt/aedt_logger.py [hunk 3]
```diff
@@ -590,9 +591,8 @@ def _log_on_dekstop(self, message_type, message_text, level=None, proj_name=None
                 des_name = des_name[des_name.find(";") + 1 :]
             try:
                 self._desktop.AddMessage(proj_name, des_name, message_type, message_text)
-
             except Exception:  # pragma: no cover
-                pass
+                self._global.info("Failed in adding desktop message")
```
Comment (PipKat) [inline pyaedt/aedt_logger.py:595]:
```suggestion
                self._global.info("Failed in adding desktop message.")
```
Conclude with period.

File: pyaedt/aedt_logger.py [hunk 4]
```diff
@@ -660,7 +660,7 @@ def clear_messages(self, proj_name=None, des_name=None, level=2):
         try:
             self._desktop.ClearMessages(proj_name, des_name, level)
         except Exception:  # pragma: no cover
-            pass
+            self._global.info("Failed in clearing desktop messages")
```
Comment (PipKat) [inline pyaedt/aedt_logger.py:663]:
```suggestion
            self._global.info("Failed in clearing desktop messages.")
```

File: pyaedt/application/Analysis.py [hunk 5]
```diff
@@ -285,7 +285,8 @@ def active_setup(self):
     def active_setup(self, setup_name):
         setup_list = self.existing_analysis_setups
         if setup_list:
-            assert setup_name in setup_list, "Invalid setup name {}".format(setup_name)
+            if setup_name not in setup_list:
+                raise ValueError("Invalid setup name {}".format(setup_name))
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:289]: Message strings must have a vern and concluding period. Rephrase: "Setup name {}".format(setup_name) is invalid.")

File: pyaedt/application/Analysis.py [hunk 6]
```diff
@@ -285,7 +285,8 @@ def active_setup(self):
...
             self._setup = setup_name
         else:
             raise AttributeError("No setup defined")
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:292]:
```suggestion
            raise AttributeError("No setup is defined.")
```

File: pyaedt/application/Analysis.py [hunk 7]
```diff
@@ -945,19 +946,18 @@ def _get_native_data(self):
...
-                                ds,
-                            )
-                        )
+                        component_type = ds["NativeComponentDefinitionProvider"]["Type"]
+                        component_name = ds["BasicComponentInfo"]["ComponentName"]
+                        native_component_object = NativeComponentObject(self, component_type, component_name, ds)
+                        boundaries.append(native_component_object)
                 except Exception:
-                    pass
+                    msg = "Failed in adding native component object"
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:956]:
```suggestion
                    msg = "Failed in adding native component object."
```

File: pyaedt/application/Analysis.py [hunk 8]
```diff
@@ -945,19 +946,18 @@ def _get_native_data(self):
...
+                    msg_end = "." if component_name == "undefined" else "(named {}).".format(component_name)
+                    self.logger.debug(msg + msg_end)
         except Exception:
-            pass
+            self.logger.debug("Failed in adding native component object.")
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:960]:
```suggestion
            self.logger.debug("Failed to add native component object.")
```

File: pyaedt/application/Analysis.py [hunk 9]
```diff
@@ -1771,7 +1770,7 @@ def analyze_setup(
                     self.set_registry_key(r"Desktop/ActiveDSOConfigurations/" + self.design_type, config_name)
                     set_custom_dso = True
                 except Exception:
-                    pass
+                    self.logger.info("Failed in setting registry from file {}".format(target_name))
         if not name:
             try:
                 self.logger.info("Solving all design setups")
```
Comment (PipKat) [inline pyaedt/application/Analysis.py:1776]:
```suggestion
                self.logger.info("Solving all design setups.")
```

File: pyaedt/application/Analysis3D.py [hunk 10]
```diff
@@ -1035,7 +1035,8 @@ def flatten_3d_components(self, component_name=None, purge_history=True, passwor
             if isinstance(component_name, str):
                 component_name = [component_name]
             for cmp in component_name:
-                assert cmp in self.modeler.user_defined_component_names, "Component Definition not found."
+                if cmp not in self.modeler.user_defined_component_names:
+                    raise ValueError("Component definition not found for '{}'.".format(cmp))
```
Comment (PipKat) [inline pyaedt/application/Analysis3D.py:1039]:
```suggestion
                    raise ValueError("Component definition was not found for '{}'.".format(cmp))
```

File: pyaedt/application/Design.py [hunk 11]
```diff
@@ -516,8 +516,9 @@ def project_properties(self):
             try:
                 settings._project_properties[os.path.normpath(self.project_file)] = load_entire_aedt_file(file_path)
             except Exception:
-                pass
-            self._logger.info("aedt file load time {}".format(time.time() - start))
+                self._logger.info("Failing in loading AEDT file.")
+            else:
+                self._logger.info("aedt file load time {}".format(time.time() - start))
```
Comment (PipKat) [inline pyaedt/application/Design.py:521]:
```suggestion
                self._logger.info("AEDT file load time {}".format(time.time() - start))
```
Make this a complete sentence with a verb and concluding punctuation. I/m not sure how the current message renders.

File: pyaedt/application/Design.py [hunk 12]
```diff
@@ -2212,7 +2212,7 @@ def _get_boundaries_data(self):
                                 )
                             )
                 except Exception:
-                    pass
+                    self.logger.debug("Failing to retrieve boundary data from BoundarySetup.")
```
Comment (PipKat) [inline pyaedt/application/Design.py:2215]:
```suggestion
                    self.logger.debug("Failed to retrieve boundary data from `BoundarySetup`.")
```

File: pyaedt/application/Design.py [hunk 13]
```diff
@@ -2230,7 +2230,7 @@ def _get_boundaries_data(self):
                             )
                         )
                 except Exception:
-                    pass
+                    self.logger.debug("Failing to retrieve boundary data from MaxwellParameterSetup.")
```
Comment (PipKat) [inline pyaedt/application/Design.py:2233]:
```suggestion
                    self.logger.debug("Failed to retrieve boundary data from 'MaxwellParameterSetup'.")
```

File: pyaedt/application/Design.py [hunk 14]
```diff
@@ -2248,7 +2248,7 @@ def _get_boundaries_data(self):
                                 )
                             )
                     except Exception:
-                        pass
+                        self.logger.debug("Failing to retrieve boundary data from ModelSetup.")
```
Comment (PipKat) [inline pyaedt/application/Design.py:2251]:
```suggestion
                        self.logger.debug("Failed to retrieve boundary data from 'ModelSetup'.")
```

File: pyaedt/application/Design.py [hunk 15]
```diff
@@ -2327,7 +2327,7 @@ def _get_project_datasets(self):
                 ]
                 datasets[ds] = self._get_ds_data(ds, data)
         except Exception:
-            pass
+            self.logger.debug("Failing to retrieve project data sets.")
```
Comment (PipKat) [inline pyaedt/application/Design.py:2330]:
```suggestion
            self.logger.debug("Failed to retrieve project data sets.")
```

File: pyaedt/application/Design.py [hunk 16]
```diff
@@ -2339,7 +2339,7 @@ def _get_design_datasets(self):
                 data = self.design_properties["ModelSetup"]["DesignDatasets"]["DatasetDefinitions"][ds]["Coordinates"]
                 datasets[ds] = self._get_ds_data(ds, data)
         except Exception:
-            pass
+            self.logger.debug("Failing to retrieve design data sets.")
```
Comment (PipKat) [inline pyaedt/application/Design.py:2342]:
```suggestion
            self.logger.debug("Failed to retrieve design data sets.")
```

File: pyaedt/application/Design.py [hunk 17]
```diff
@@ -3264,9 +3264,9 @@ def insert_design(self, design_name=None, solution_type=None):
         )
 
     def _insert_design(self, design_type, design_name=None):
-        assert design_type in self.design_solutions.design_types, "Invalid design type for insert: {}".format(
-            design_type
-        )
+        if design_type not in self.design_solutions.design_types:
+            raise ValueError("Invalid design type for insert: {}".format(design_type))
```
Comment (PipKat) [inline pyaedt/application/Design.py:3268]: Reformat so it reads "Design type of insert x is invalid."

File: pyaedt/aedt_logger.py (hunk 3 above)
Comment (SMoraisAnsys) [inline pyaedt/aedt_logger.py:595]:
```suggestion
                self._global.info("Failed to add desktop message.")
```

File: pyaedt/aedt_logger.py (hunk 4 above)
Comment (SMoraisAnsys) [inline pyaedt/aedt_logger.py:663]:
```suggestion
            self._global.info("Failed to clear desktop messages.")
```

File: pyaedt/application/Analysis.py (hunk 5 above)
Comment (SMoraisAnsys) [inline pyaedt/application/Analysis.py:289]:
```suggestion
                raise ValueError("Setup name {} is invalid.".format(setup_name))
```

File: pyaedt/application/Analysis.py [hunk 18]
```diff
@@ -706,7 +707,7 @@ def export_results(
                 self.post.oreportsetup.ExportToFile(str(report_name), export_path)
                 self.logger.info("Export Data: {}".format(export_path))
             except Exception:
-                pass
+                self.logger.info("Failed in exporting to file.")
```
Comment (SMoraisAnsys) [inline pyaedt/application/Analysis.py:710]:
```suggestion
                self.logger.info("Failed to export to file.")
```

File: pyaedt/application/Analysis.py [hunk 19]
```diff
@@ -1771,7 +1770,7 @@ def analyze_setup(
                     self.set_registry_key(r"Desktop/ActiveDSOConfigurations/" + self.design_type, config_name)
                     set_custom_dso = True
                 except Exception:
-                    pass
+                    self.logger.info("Failed in setting registry from file {}".format(target_name))
```
Comment (SMoraisAnsys) [inline pyaedt/application/Analysis.py:1773]:
```suggestion
                    self.logger.info("Failed to set registry from file {}.".format(target_name))
```

File: pyaedt/application/Design.py [hunk 20]
```diff
@@ -396,7 +396,7 @@ def boundaries(self):
                 if k not in self._boundaries:
                     self._boundaries[k] = BoundaryObject(self, k, boundarytype=v)
         except Exception:
-            pass
+            self.logger.info("Failing in designing boundary object.")
```
Comment (SMoraisAnsys) [inline pyaedt/application/Design.py:399]:
```suggestion
            self.logger.info("Failed to design boundary object.")
```

File: pyaedt/application/Design.py [hunk 21]
```diff
@@ -516,8 +516,9 @@ def project_properties(self):
             try:
                 settings._project_properties[os.path.normpath(self.project_file)] = load_entire_aedt_file(file_path)
             except Exception:
-                pass
-            self._logger.info("aedt file load time {}".format(time.time() - start))
+                self._logger.info("Failing in loading AEDT file.")
```
Comment (SMoraisAnsys) [inline pyaedt/application/Design.py:519]:
```suggestion
                self._logger.info("Failed to load AEDT file.")
```

File: pyaedt/application/Design.py (hunk 11 above)
Comment (SMoraisAnsys) [inline pyaedt/application/Design.py:521]:
```suggestion
                self._logger.info("Time to load AEDT file: {}.".format(time.time() - start))
```

File: pyaedt/application/Design.py (hunk 17 above)
Comment (SMoraisAnsys) [inline pyaedt/application/Design.py:3268]:
```suggestion
            raise ValueError("Design type of insert '{}' is invalid.".format(design_type))
```

**Pull request merged**